hugr/src/extension.rs

```diff
@@ -177,18 +177,22 @@ pub enum SignatureError {
 
 /// Concrete instantiations of types and operations defined in extensions.
 trait CustomConcrete {
+    /// The identifier type for the concrete object.
+    type Identifier;
```

Comment:
I left this associated type even though it always gets instantiated with `SmolStr`.

It doesn't change the way we use this trait but documents a bit better the behaviour.